actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -311,8 +311,9 @@ def button_to(name = nil, options = nil, html_options = nil, &block)
         form_options[:action] = url
         form_options[:'data-remote'] = true if remote
 
-        request_token_tag = if form_method == 'post'
-          token_tag(nil, form_options: form_options)
+        token_tag_method = method == 'delete' ? 'delete' : form_method
+        request_token_tag = if (form_method == 'post' || method == 'delete')
+          token_tag(nil, form_options: form_options.merge(method: token_tag_method))
```

Comment:
Actually, those 3 lines are heavily condensed for no reason. Prefer some incidental duplication with:

```ruby
request_token_tag = \
  if form_method == 'post'
    token_tag(nil, form_options: form_options)
  elsif method == 'delete'
    token_tag(nil, form_options: form_options.merge(method: 'delete'))
  else
    ''
  end
```